src/finch/util/config.py

```diff
@@ -0,0 +1,68 @@
+import os
+import json
+
+"""
+Finch Configuration Module
+
+This module manages configuration settings for the Finch application. 
+Finch stores its settings and data in the `FINCH_PATH` directory, which 
+defaults to `~/.finch` but can be customized using the `FINCH_PATH` 
+environment variable.
+
+Configuration details:
+- Settings are stored in a `config.json` file within the `FINCH_PATH` directory.
+- Values can be set via environment variables, the `config.json` file, 
+    or the `set_config` function.
+- Configuration values are loaded automatically when the module is imported 
+    and can be accessed using the `get_config` function.
+
+Use this module to easily manage and retrieve Finch-specific settings.
+"""
+
+depot_dir = os.path.realpath(os.path.expanduser(os.getenv('FINCH_PATH', os.path.join("~", ".finch"))))
+
+default_config = {
+    "FINCH_CACHE_PATH": os.path.join(depot_dir, "cache"),
```

Comment:
I'd add the package version here -- cache can easily be poisoned by using the wrong version!

Reply:
Ah, yes. Version is one of many things we want to add to the cache. We can also use the arguments to the function as cache keys, but I think having the version in there is extra helpful, I'll add it.

Reply:
I can't think of a case where we wouldn't want to use the version as a cache key. I'm wondering now if we want to use a hash of the finch-tensor-lite directory as a cache key.

Reply:
Done, I skipped the directory hash but we should consider it.